From bfd9cd9e603ef0d0e593d4432048bffc2acfeb7c Mon Sep 17 00:00:00 2001
From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date: Fri, 25 Nov 2016 19:42:27 +0200
Subject: [PATCH] Allow build with uClibc

Basically this replaces type definitions in <uchar.h>.

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
[Erico: rebase to 38]
Signed-off-by: Erico Nunes <nunes.erico@gmail.com>
---
 src/export.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/export.c b/src/export.c
index db5e637..21c5617 100644
--- a/src/export.c
+++ b/src/export.c
@@ -9,10 +9,16 @@
 #include <inttypes.h>
 #include <stdint.h>
 #include <stdlib.h>
-#include <uchar.h>
 
 #include "efivar.h"
 
+#ifdef __UCLIBC__
+typedef int_least16_t char16_t;
+typedef int_least32_t char32_t;
+#else
+#include <uchar.h>
+#endif
+
 #define EFIVAR_MAGIC 0xf3df1597u
 
 #define ATTRS_UNSET 0xa5a5a5a5a5a5a5a5
-- 
2.37.3

